packages/clerk-js/src/core/clerk.ts

```diff
@@ -543,84 +541,121 @@ export default class Clerk implements ClerkInterface {
+      const devBrowserJwt = this.#devBrowserHandler?.getDevBrowserJWT();
+      if (!devBrowserJwt) {
+        return clerkMissingDevBrowserJwt();
+      }
+
+      toURL.hash = setSearchParameterInHash({
+        hash: toURL.hash,
+        paramName: DEV_BROWSER_SSO_JWT_PARAMETER,
+        paramValue: devBrowserJwt,
+      });
+    }
+
+    return toURL.href;
+  }
+
+  public buildSignInUrl(options?: RedirectOptions): string {
     const opts: RedirectOptions = {
       ...options,
       redirectUrl: options?.redirectUrl || window.location.href,
     };
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
     const { signInUrl } = this.#environment.displayConfig;
-    const url = appendAsQueryParams(signInUrl, opts);
-    return this.redirectWithAuth(url);
-  };
+    return this.buildUrlWithAuth(appendAsQueryParams(signInUrl, opts));
+  }
 
-  public redirectToSignUp = async (options?: RedirectOptions): Promise<unknown> => {
+  public buildSignUpUrl(options?: RedirectOptions): string {
     const opts: RedirectOptions = {
       ...options,
       redirectUrl: options?.redirectUrl || window.location.href,
     };
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
     const { signUpUrl } = this.#environment.displayConfig;
-    const url = appendAsQueryParams(signUpUrl, opts);
-    return this.redirectWithAuth(url);
-  };
+    return this.buildUrlWithAuth(appendAsQueryParams(signUpUrl, opts));
+  }
 
-  public redirectToUserProfile = async (): Promise<unknown> => {
+  public buildUserProfileUrl(): string {
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
-    return this.redirectWithAuth(this.#environment.displayConfig.userProfileUrl);
-  };
+    return this.buildUrlWithAuth(this.#environment.displayConfig.userProfileUrl);
+  }
 
-  public redirectToHome = async (): Promise<unknown> => {
+  public buildHomeUrl(): string {
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
-    return this.redirectWithAuth(this.#environment.displayConfig.homeUrl);
-  };
+    return this.buildUrlWithAuth(this.#environment.displayConfig.homeUrl);
+  }
+
+  public buildCreateOrganizationUrl(): string {
+    if (!this.#environment || !this.#environment.displayConfig) {
+      return '';
+    }
+    return this.buildUrlWithAuth(this.#environment.displayConfig.createOrganizationUrl);
+  }
+
+  public buildOrganizationProfileUrl(): string {
+    if (!this.#environment || !this.#environment.displayConfig) {
+      return '';
+    }
+    return this.buildUrlWithAuth(this.#environment.displayConfig.organizationProfileUrl);
+  }
 
   public redirectWithAuth = async (to: string): Promise<unknown> => {
-    if (!inBrowser()) {
-      return;
+    if (inBrowser()) {
+      return this.navigate(this.buildUrlWithAuth(to));
```

Comment:
:upside_down_face: Similar comment about repetition, the `inBrowser()` check is repeated in all functions below.